cdap-runtime-ext-dataproc/src/main/java/co/cask/cdap/runtime/spi/provisioner/dataproc/DataprocProvisioner.java

```diff
@@ -53,12 +58,85 @@
     "Google Cloud Dataproc is a fast, easy-to-use, fully-managed cloud service for running Apache Spark and Apache " +
       "Hadoop clusters in a simpler, more cost-efficient way.");
   private static final String CLUSTER_PREFIX = "cdap-";
+  // keys and values cannot be longer than 63 characters
+  // keys and values can only contain lowercase letters, numbers, underscores, and dashes
+  // keys must start with a lowercase letter
+  // keys cannot be empty
+  private static final Pattern LABEL_KEY_PATTERN = Pattern.compile("^[a-z][a-z0-9_-]{0,62}$");
+  private static final Pattern LABEL_VAL_PATTERN = Pattern.compile("^[a-z0-9_-]{0,63}$");
+  private Map<String, String> systemLabels;
 
   @Override
   public ProvisionerSpecification getSpec() {
     return SPEC;
   }
 
+  @Override
+  public void initialize(ProvisionerSystemContext systemContext) {
+    Map<String, String> labels = new HashMap<>();
+    // dataproc only allows label values to be lowercase letters, numbers, or dashes
+    String cdapVersion = systemContext.getCDAPVersion().toLowerCase();
+    cdapVersion = cdapVersion.replaceAll("\\.", "_");
+    labels.put("cdap-version", cdapVersion);
+
+    String extraLabelsStr = systemContext.getProperties().get("labels");
```

Comment:
Seems like we only care about the labels in the properties? Is there anything else can be set using the property?

Reply:
we don't currently have another use case for this.